crates/sequencer/src/transaction.rs

```diff
@@ -0,0 +1,93 @@
+use std::sync::Arc;
+
+use blockifier::transaction::{
+    account_transaction::AccountTransaction,
+    transaction_execution::Transaction as ExecutionTransaction,
+};
+use starknet::core::crypto::compute_hash_on_elements;
+use starknet::core::types::{BroadcastedTransaction, FieldElement};
+use starknet_api::core::{ContractAddress, Nonce, PatriciaKey};
+use starknet_api::hash::{StarkFelt, StarkHash};
+use starknet_api::transaction::InvokeTransaction;
+use starknet_api::transaction::{
+    Calldata, Fee, InvokeTransactionV1, TransactionHash, TransactionSignature,
+};
+
+#[derive(Debug)]
+pub struct StarknetTransaction(BroadcastedTransaction);
+
+impl StarknetTransaction {
+    pub fn new(transaction: BroadcastedTransaction) -> Self {
+        Self(transaction)
+    }
+
+    pub fn try_into_execution_transaction(
+        self,
+        chain_id: FieldElement,
+    ) -> Result<ExecutionTransaction, eyre::Error> {
+        match self.0 {
+            BroadcastedTransaction::Invoke(invoke) => Ok(ExecutionTransaction::AccountTransaction(
+                AccountTransaction::Invoke(InvokeTransaction::V1(InvokeTransactionV1 {
+                    transaction_hash: TransactionHash(Into::<StarkHash>::into(
+                        Into::<StarkFelt>::into(compute_transaction_hash(
+                            invoke.sender_address,
+                            &invoke.calldata,
+                            invoke.max_fee,
+                            chain_id,
+                            invoke.nonce,
+                        )),
+                    )),
+                    max_fee: Fee(invoke.max_fee.try_into()?),
+                    signature: TransactionSignature(
+                        invoke
+                            .signature
+                            .into_iter()
+                            .map(Into::<StarkFelt>::into)
+                            .collect(),
+                    ),
+                    nonce: Nonce(invoke.nonce.try_into()?),
+                    sender_address: ContractAddress(TryInto::<PatriciaKey>::try_into(Into::<
+                        StarkHash,
+                    >::into(
+                        Into::<StarkFelt>::into(invoke.sender_address),
+                    ))?),
+                    calldata: Calldata(Arc::new(
+                        invoke
+                            .calldata
+                            .into_iter()
+                            .map(Into::<StarkFelt>::into)
+                            .collect(),
+                    )),
+                })),
+            )),
+            // TODO: Add support for other transaction types.
+            _ => Err(eyre::eyre!("Unsupported transaction type")),
+        }
+    }
+}
+
+const PREFIX_INVOKE: FieldElement = FieldElement::from_mont([
```

Comment:
while this is (probably) true, keeping here a hex string and doing would be more explicit

https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/transactions/#v1_hash_calculation

Reply:
this is copied from starknet-rs but I can udpate